From 861761252dcf07e3c08f87ac39e3f0eb003d1fd1 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@marvell.com>
Date: Tue, 28 Jan 2020 18:54:00 -0800
Subject: [PATCH 06/21] drivers: marvell: octeontx2: sdei_ghes: add LMC

commit a1073be9953d1f0158e591d77aa601ea052ec652 from
git@git.assembla.com:cavium/WindRiver.linux.git

Enable MSIX handling on LMC devices too.

Change-Id: Ic123fe2a8859149e029be565955a8e23926782ba
Signed-off-by: Peter Swain <pswain@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/22599
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Rick Farrington <Ricardo.Farrington@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
index 1cc7213c3d06..eae0c4e0fca9 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
@@ -40,9 +40,11 @@ static struct acpi_table_hest *hest;
 static struct otx2_ghes_event *event_list;
 
 #define PCI_VENDOR_ID_CAVIUM            0x177d
+#define PCI_DEVICE_ID_OCTEONTX2_LMC     0xa022
 #define PCI_DEVICE_ID_OCTEONTX2_MCC     0xa070
 #define PCI_DEVICE_ID_OCTEONTX2_MDC     0xa073
 static const struct pci_device_id sdei_ghes_otx2_pci_tbl[] = {
+	{ PCI_DEVICE(PCI_VENDOR_ID_CAVIUM, PCI_DEVICE_ID_OCTEONTX2_LMC) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_CAVIUM, PCI_DEVICE_ID_OCTEONTX2_MCC) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_CAVIUM, PCI_DEVICE_ID_OCTEONTX2_MDC) },
 	{ 0, },
@@ -62,7 +64,6 @@ static int sdei_ghes_callback(u32 event_id, struct pt_regs *regs, void *arg)
 
 	ghes_drv = arg;
 
-
 	for (idx = 0; idx < ghes_drv->event_count; idx++) {
 		event = &ghes_drv->event_list[idx];
 		if (event->id != event_id)
@@ -579,6 +580,7 @@ static int __init sdei_ghes_driver_init(void)
 	}
 
 	platform_driver_register(&sdei_ghes_drv);
+
 	/* Enable MSIX for devices whose [secure] IRQ's we control.
 	 * These IRQs have been initialized by ATF.
 	 * This is required due to an errata against
@@ -601,4 +603,3 @@ device_initcall(sdei_ghes_driver_init);
 MODULE_DESCRIPTION("OcteonTX2 SDEI GHES Driver");
 MODULE_LICENSE("GPL v2");
 MODULE_ALIAS("platform:" DRV_NAME);
-
-- 
2.17.1

